infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/TolerationsProvisioner.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import static java.util.Collections.emptyList;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.Toleration;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+
+/** Provisions tolerations into workspace pod spec. */
+public class TolerationsProvisioner implements ConfigurationProvisioner {
+
+  private final List<Toleration> tolerations;
+
+  @Inject
+  public TolerationsProvisioner(
+      @Nullable @Named("che.workspace.pod.tolerations") String tolerationsProperty)
+      throws
+          JsonProcessingException { // TODO: What kind of exception should I throw here?  Code the
```

Comment:
I need some advice on how to handle the JsonProcessingException that could be raised when there is an error in the json string.

Reply:
I believe the best choice here is to throw `ConfigurationException`, as we do in e.g. [`KuberentesNamespaceFactory`](https://github.com/eclipse/che/blob/9e3da969757f65e680d8dc9a7d16c7a0d477fb4e/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java#L152)